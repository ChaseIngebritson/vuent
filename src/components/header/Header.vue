<script>
export default {
  name: 'VntHeader',

  render(createElement) {
    let header = null;
    let subheader = null;

    if (this.$slots.header || this.$slots.default) {
      header = createElement('div', {
        attrs: {
          class: 'vnt-header__header'
        }
      }, this.$slots.header || this.$slots.default);
    }

    if (this.$slots.subheader) {
      subheader = createElement('div', {
        attrs: {
          class: 'vnt-header__subheader'
        }
      }, this.$slots.subheader);
    }

    return createElement('div', {
      attrs: {
        class: 'vnt-header'
      }
    }, [header, subheader]);
  }
};
</script>

<style lang="scss">
@import '../../scss/mixins/component';

.vnt-header {
  @include component-base();

  color: #000100;
  font-weight: 300;
}

%clear-margin {
  margin: 0.2em 0;

  & > * {
    margin: 0;
  }
}

.vnt-header__header {
  @extend %clear-margin;

  font-size: 46px;
  line-height: 62px;
}

.vnt-header__subheader {
  @extend %clear-margin;

  font-size: 34px;
  line-height: 46px;
}
</style>
