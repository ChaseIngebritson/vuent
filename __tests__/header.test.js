import { mount } from 'vue-test-utils';
import Header from '@/components/header';

describe('Header', () => {
  let wrapper;

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(Header);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  test('can have only main header content set', () => {
    wrapper = mount(Header, {
      slots: {
        default: 'Header'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('can have main header content set using named slot', () => {
    wrapper = mount(Header, {
      slots: {
        header: 'Header'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('can have both header and subheader content set', () => {
    wrapper = mount(Header, {
      slots: {
        header: 'Header',
        subheader: 'Subheader'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('can have only subheader content set', () => {
    wrapper = mount(Header, {
      slots: {
        subheader: 'Subheader'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

});
