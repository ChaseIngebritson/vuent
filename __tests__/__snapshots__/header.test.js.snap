// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header by default renders correctly 1`] = `<div class="vnt-header"></div>`;

exports[`Header can have both header and subheader content set 1`] = `
<div class="vnt-header">
    <div class="vnt-header__header">Header</div>
    <div class="vnt-header__subheader">Subheader</div>
</div>
`;

exports[`Header can have main header content set using named slot 1`] = `
<div class="vnt-header">
    <div class="vnt-header__header">Header</div>
</div>
`;

exports[`Header can have only main header content set 1`] = `
<div class="vnt-header">
    <div class="vnt-header__header">Header</div>
</div>
`;

exports[`Header can have only subheader content set 1`] = `
<div class="vnt-header">
    <div class="vnt-header__subheader">Subheader</div>
</div>
`;
